### PR TITLE
2016.11 to 2017.7 merge.

### PR DESCRIPTION
**Pull request by dubb-b**
Merge forward of 2016.11 to 2017.7

### CODE REVIEW

File: python/boto.sls [hunk 1]
```diff
@@ -12,9 +12,7 @@ boto:
     {%- if salt['config.get']('pip_target', None)  %}
     - target: {{ salt['config.get']('pip_target') }}
     {%- endif %}
-    - index_url: https://pypi.c7.saltstack.net/simple
-    - extra_index_url: https://pypi.python.org/simple
-{% if grains['os'] not in ('Windows',) %}
```
Comment (s0undt3ch) [inline python/boto.sls]: The removed comma should be readded. On all occurrences in this PR.

File: python/moto.sls [hunk 2]
```diff
@@ -15,9 +15,7 @@ moto:
     {%- if salt['config.get']('virtualenv_path', None)  %}
     - bin_env: {{ salt['config.get']('virtualenv_path') }}
     {%- endif %}
-    - index_url: https://pypi.c7.saltstack.net/simple
-    - extra_index_url: https://pypi.python.org/simple
-{%- if grains['os'] not in ('Windows',) %}
+{%- if grains['os'] not in ('Windows') %}
```
Comment (s0undt3ch) [inline python/moto.sls]: Missed this comma :)

File: python/supervisor.sls [hunk 3]
```diff
@@ -6,7 +6,6 @@ supervisor:
     {%- if salt['config.get']('virtualenv_path', None)  %}
     - bin_env: {{ salt['config.get']('virtualenv_path') }}
     {%- endif %}
-    - index_url: https://pypi.c7.saltstack.net/simple
-    - extra_index_url: https://pypi.python.org/simple
+{% if grains['os'] not in ('Windows',) %}
```
Comment (gtmanfred) [inline python/supervisor.sls]: This does not have a close endif.

**Comment by gtmanfred** [comment]
This breaks
```
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/lsb_release.sls' using 'yaml' renderer: 0.00105714797974
[DEBUG   ] LazyLoaded highstate.output
local:
    Data failed to compile:
----------
    Rendering SLS 'base:python.supervisor' failed: Jinja syntax error: Unexpected end of template. Jinja was looking for the following tags: 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.; line 9

---
[...]
supervisor:
  pip2.installed:
    {%- if salt['config.get']('virtualenv_path', None)  %}
    - bin_env: {{ salt['config.get']('virtualenv_path') }}
    {%- endif %}
{% if grains['os'] not in ('Windows',) %}    <======================
    - require:
      - cmd: pip-install
---
----------
    Rendering SLS 'base:python.psutil' failed: Jinja syntax error: Unexpected end of template. Jinja was looking for the following tags: 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.; line 23

---
[...]
      - pkg: python-dev
      {%- endif %}
      {%- if grains['os_family'] not in ('FreeBSD', 'Gentoo', 'Windows') %}
        {#- FreeBSD always ships with gcc #}
      - pkg: gcc
      {%- endif %}    <======================
      - cmd: pip-install
---
```

**Pull request merged**